src/Psalm/Internal/Analyzer/MethodComparator.php

```diff
@@ -457,10 +456,8 @@ private static function compareMethodParams(
                     $implementer_param->location->file_path,
                 )
             ) {
-                if (!$guide_classlike_storage->user_defined && $i === 0 && count($guide_method_storage->params) < 2) {
-                    // if it's third party defined and a single arg, renaming is unnecessary
-                    // if we still want to psalter it, move this if and change the else below to elseif
-                } elseif ($config->allow_named_arg_calls
+                // even if it's just a single arg, it needs to be renamed in case it's called with a single named arg
+                if ($config->allow_named_arg_calls
```

Comment:
I'm curious, what was the original logic for allowing mismatches for the single parameter methods?

Reply:
Hmm, strange. `git praise` points to your commit:
https://github.com/vimeo/psalm/commit/0252bbbd7c?diff=split&w=0#diff-c70a8399e3804f1bb260231f6af972513c1ca05d21e3f4221ae35ad5a9d4e6c0R459